themes/ddbasic/sass/components/ding-carousel.scss

```diff
@@ -232,3 +232,17 @@
     }
   }
 }
+
+.ding-carousel-actions {
+  position: relative;
+  width: calc(100% - 166px);
+  bottom: 84px;
+  .more-link {
+    margin-right: 15px;
+  }
+  form {
```

Comment:
Why would there be a form inside `.ding-carousel-actions`? There does not seem to be any where it is defined in  `template_preprocess_ding_list_list()` as a part of this PR.

Reply:
Do you have a link to this? [I cannot see it](https://github.com/ding2/ding2/blob/master/modules/p2/ding_list/theme/theme.inc#L189-L201).

Reply:
Added here:
https://github.com/ding2/ding2/pull/1213/commits/9444088b70a133f4d8410979cdce6dbdebcaff80#diff-32b73877212e335408ce681993ee780dR196